Assets/Scripts/Localization/LocalizationManager.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections.Generic;
+using PotentialRobot.Localization.Components;
+using UnityEngine;
+
+namespace PotentialRobot.Localization
+{
+    public class LocalizationManager
+    {
+        public const string c_currentLanguageKey = "CurrentLanguage";
+        private static LocalizationManager _instance;
+        public static LocalizationManager Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new LocalizationManager();
+                    _instance.Initialize();
+                }
+                return _instance;
+            }
+        }
+
+        public delegate void LanguageChanged(string language);
+        private LanguageChanged _onLanguageChanged;
+        private Dictionary<string, string> _localizations;
+        private string _previousLanguage;
+        public string CurrentLanguage { get; private set; }
+
+        public void Initialize()
+        {
+            if (PlayerPrefs.HasKey("CurrentLanguage"))
+                CurrentLanguage = PlayerPrefs.GetString("CurrentLanguage");
+            else
+                SetFirstLanguage();
+
+            LoadLocalization(CurrentLanguage);
+        }
+
+        private void SetFirstLanguage()
+        {
+            CurrentLanguage = "English";
+            PlayerPrefs.SetString("CurrentLanguage", CurrentLanguage);
+        }
+
+        public void LoadLocalization(string language)
+        {
+            if (_previousLanguage == language)
+                return;
+            
+            LoadLanguageInfo(language);
+            
+            CurrentLanguage = language;
+            _previousLanguage = language;
+            _onLanguageChanged?.Invoke(language);
+        }
+
+        public void LoadLanguageInfo(string language)
+        {
+            var locAsset = Resources.Load<LocalizationAsset>($"Localization/LocalizationAsset_{language}.asset");
+            if (locAsset == null)
+            {
+                Debug.LogError("Localization error: failed to load localization file.");
```

Comment:
Consider including the name of the missing file in the error message.

Reply:
Added full path of the file.